hive-event-listeners/apiary-gluesync-listener/src/test/java/com/expediagroup/apiary/extensions/gluesync/cli/GlueSyncCliParserTest.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (C) 2018-2025 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.apiary.extensions.gluesync.cli;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.UnknownDBException;
+import org.apache.thrift.TException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GlueSyncCliParserTest {
+
+  private ByteArrayOutputStream outContent;
+  private ByteArrayOutputStream errContent;
+  private PrintStream originalOut;
+  private PrintStream originalErr;
+  private SecurityManager originalSecurityManager;
+  static {
+    // Use Mockito to mock GlueSyncCli and prevent real constructor logic
+    GlueSyncCli mock = mock(GlueSyncCli.class);
+    GlueSyncCliParser.setGlueSyncCli(mock);
+  }
+
+  @Before
+  public void setUp() {
+    // Capture System.out and System.err
+    outContent = new ByteArrayOutputStream();
```

Comment:
Consider using this library: https://stefanbirkner.github.io/system-rules/

We already us it on other places: https://github.com/ExpediaGroup/apiary-extensions/blob/main/apiary-metastore-events/sns-metastore-events/apiary-metastore-listener/pom.xml#L56

Reply:
done, thanks